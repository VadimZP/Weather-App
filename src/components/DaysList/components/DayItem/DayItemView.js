import React from 'react'
import PropTypes from 'prop-types'

DayItemView.propTypes = {
    days: PropTypes.arrayOf(PropTypes.object),
    date: PropTypes.string,
    dayOfTheWeek: PropTypes.string,
    minTemp: PropTypes.number,
    maxTemp: PropTypes.number,
    weatherImg: PropTypes.string,
    id: PropTypes.number,
    getWeather: PropTypes.func,
}

export default function DayItemView(props) {
    return (
        <div
            className="day"
            id={props.id}
            role="button"
            tabIndex="0"
            onClick={() => props.getWeather(props.days)}
            onKeyDown={() => props.getWeather(props.days)}
            onFocus={() => props.getWeather(props.days)}
        >
            <p className="weekday"><b>{props.dayOfTheWeek}</b></p>
            <img
                src={`http://openweathermap.org/img/w/${props.weatherImg}.png`}
                style={{ width: 50, height: 50 }}
                alt="weather"
            />
            <p className="temperature">
                <b>Max: </b>{props.maxTemp}°C <br /><b>Min: </b>{props.minTemp}°C
            </p>
            <p className="date">{props.date}</p>
        </div>
    )
}
