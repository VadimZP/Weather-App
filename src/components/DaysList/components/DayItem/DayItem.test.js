import React from 'react'
import renderer from 'react-test-renderer'

import DayItemView from './DayItemView'

test('Return DayItem', () => {
    const component = renderer.create(<DayItemView
        onClick={() => {}}
        days={[{}, {}]}
        id={1}
        key={2}
        weatherImg="rain"
        date="2017-12-30"
        dayOfTheWeek="Monday"
        minTemp={-16}
        maxTemp={+30}
    />)
    const DayItemViewTree = component.toJSON()

    expect(DayItemViewTree).toMatchSnapshot()
})
