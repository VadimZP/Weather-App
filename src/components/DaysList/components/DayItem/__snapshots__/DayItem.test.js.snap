// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Return DayItem 1`] = `
<div
  className="day"
  id={1}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <p
    className="weekday"
  >
    <b>
      Monday
    </b>
  </p>
  <img
    alt="weather"
    src="http://openweathermap.org/img/w/rain.png"
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  />
  <p
    className="temperature"
  >
    <b>
      Max: 
    </b>
    30
    °C 
    <br />
    <b>
      Min: 
    </b>
    -16
    °C
  </p>
  <p
    className="date"
  >
    2017-12-30
  </p>
</div>
`;
